atomicfu/src/commonMain/kotlin/kotlinx/atomicfu/AtomicFU.common.kt

```diff
@@ -10,51 +10,98 @@ import kotlin.js.JsName
 import kotlin.internal.InlineOnly
 import kotlinx.atomicfu.TraceBase.None
 import kotlin.reflect.KProperty
-import kotlin.reflect.KProperty0
 
+/**
+ * Creates atomic reference with a given [initial] value and a [trace] object to trace modifications of the value.
+ *
+ * It can only be used to initialize a private or internal read-only property, like this:
+ *
+ * ```
+ * private val f = atomic<Type>(initial, trace)
+ * ```
+ */
+public expect fun <T> atomic(initial: T, trace: TraceBase = None): AtomicRef<T>
+
+// Binary compatibility with IR, should be removed with Kotlin 1.5 release
 /**
  * Creates atomic reference with a given [initial] value.
  *
- * It can only be used in initialize of private read-only property, like this:
+ * It can only be used to initialize a private or internal read-only property, like this:
  *
  * ```
  * private val f = atomic<Type>(initial)
  * ```
  */
-public expect fun <T> atomic(initial: T, trace: TraceBase = None): AtomicRef<T>
+public expect fun <T> atomic(initial: T): AtomicRef<T>
 
+/**
+ * Creates atomic [Int] with a given [initial] value and a [trace] object to trace modifications of the value.
```

Comment:
I'd suggest adding a link to the trace so it will be easier to understand what trace is:
`[trace modifications][Trace]`

Reply:
Also, while we are here, `traceFormatDefault` doc states "the default **trade** string formatter". Probably meant to be "trace"